File: frontend/src/types/matching.ts
import type { CapacityStatus } from '@/types/capacity'
import type { SkillLevel } from '@/types/skill'

export interface SkillRequirementMatch {
  skill_id: number
  skill_name: string
  required_level: SkillLevel
  employee_level: SkillLevel | null
  met: boolean
}

export interface EmployeeProjectMatch {
  employee_id: number
  employee_name: string
  requirements_met: number
  requirements_total: number
  matched_requirements: SkillRequirementMatch[]
  unmet_requirements: SkillRequirementMatch[]
  allocated_percent: number
  available_percent: number
  capacity_status: CapacityStatus
}

File: backend/app/schemas/matching.py
from pydantic import BaseModel

from app.models.skill import SkillLevel
from app.schemas.capacity import CapacityStatus


class SkillRequirementMatchRead(BaseModel):
    skill_id: int
    skill_name: str
    required_level: SkillLevel
    employee_level: SkillLevel | None
    met: bool


class EmployeeProjectMatchRead(BaseModel):
    employee_id: int
    employee_name: str
    requirements_met: int
    requirements_total: int
    matched_requirements: list[SkillRequirementMatchRead]
    unmet_requirements: list[SkillRequirementMatchRead]
    allocated_percent: int
    available_percent: int
    capacity_status: CapacityStatus
